### PR TITLE
More doc improvements

### PR DESCRIPTION
**Pull request by aboedo**
- Moved `Known Issues` section to the bottom
- added SwiftUI example
- added callout to `CustomerInfoStream` in `New APIs`

### CODE REVIEW

File: Documentation.docc/V4_API_Migration_guide.md [hunk 1]
```diff
@@ -279,7 +265,23 @@ if let error = error as? RevenueCat.ErrorCode {
 - `beginRefundRequestForCurrentEntitlement`: Use this method to begin a refund request for the purchase that granted the current entitlement.
 - `beginRefundRequest(forProduct:)`: Use this method to begin a refund request for a purchase, specifying the product identifier.
 - `beginRefundRequest(forEntitlement:)`: Use this method to begin a refund request for a purchase, specifying the entitlement identifier.
+- You can now use ``Purchases/customerInfoStream`` to be notified whenever there's new CustomerInfo available,
```
Comment (NachoSoto) [inline Documentation.docc/V4_API_Migration_guide.md:268]: Nit:
```suggestion
- You can now use ``Purchases/customerInfoStream`` to be notified whenever there's new ``CustomerInfo`` available, 
```

File: Purchases/Purchasing/Purchases.swift [hunk 2]
```diff
@@ -876,9 +876,27 @@ public extension Purchases {
...
     /// ```
-    /// - Note: an alternative way of getting ``CustomerInfo`` updates
-    /// is using ``PurchasesDelegate/purchases(_:receivedUpdated:)``
-    /// - Note: this method is not thread-safe.
+    /// #### SwiftUI Example:
+    /// ```swift
+    /// ContentView()
+    ///   .environmentObject(revenueCatCustomerData)
+    ///   .task {
+    ///     if let customerInfo = try? await Purchases.shared.customerInfo() {
```
Comment (NachoSoto) [inline Purchases/Purchasing/Purchases.swift:884]: This isn't needed now that the stream starts with the first value (https://github.com/RevenueCat/purchases-ios/pull/1257)

File: Documentation.docc/V4_API_Migration_guide.md [hunk 3]
```diff
@@ -13,20 +13,6 @@ It also updates the minimum deployment targets for iOS, macOS and tvOS.
 | macOS | 10.12 | 10.13 |
 | watchOS | 6.2 | 6.2 (unchanged) |
 
-## Known Issues
```
Comment (NachoSoto) [inline Documentation.docc/V4_API_Migration_guide.md:16]: 👍 much better below.

File: Purchases/Purchasing/Purchases.swift (hunk 2 above)
Comment (aboedo) [reply Purchases/Purchasing/Purchases.swift]: nice

File: Documentation.docc/V4_API_Migration_guide.md (hunk 1 above)
Comment (aboedo) [reply Documentation.docc/V4_API_Migration_guide.md]: great catch

File: Purchases/Purchasing/Purchases.swift [hunk 4]
```diff
@@ -876,9 +876,27 @@ public extension Purchases {
     ///   ...
     /// }
     /// ```
-    /// - Note: an alternative way of getting ``CustomerInfo`` updates
-    /// is using ``PurchasesDelegate/purchases(_:receivedUpdated:)``
-    /// - Note: this method is not thread-safe.
+    /// #### SwiftUI Example:
```
[the remark this answers is not in the document]
Comment (aboedo) [reply Purchases/Purchasing/Purchases.swift]: makes sense. Really this looks almost exactly the same as the regular example (once you remove the unnecessary first customerInfo), so I don't think it adds much. Killing it for now

**Review by NachoSoto — APPROVED** [review]

**Pull request merged**